src/openrct2/interface/Window.cpp

```diff
@@ -227,7 +227,11 @@ void WindowSetWindowLimit(int32_t value)
  */
 void WindowClose(WindowBase& w)
 {
-    w.OnClose();
+    if (!w.OnClose())
+    {
+        // Something's preventing this window from closing -- bail out early
+        return;
+    }
```

Comment:
I wonder whether it would be clearer from a design standpoint if we added another function instead, and made it such as:

```c++
if (w.CanClose())
{
    w.OnClose();
}
```

Then you wouldn't need to touch so many windows, reducing the PR size, and I think it would be easier for the random contributors to understand what is expected

Reply:
I also agree with this, OnClose should stay strictly an event for when the window is really going to close. It would also reduce the amount of changes needed as not all windows need to override CanClose. Also how sure are we that this function is always used to close a window?

Reply:
Alright, I'll add another `CanClose` function then.

> Also how sure are we that this function is always used to close a window?

Well, it is literally called when a window is on the cusp of being closed. One of the window event functions.

Reply:
I've seen code like w->Close() so just making sure it actually ends up preventing it. This adds a bit of fuzzyness given that when someone calls WindowClose or w->Close one would ordinarily expect the window to be closed but I can see to why we should prevent it, at least that one window. Not gonna lie not a big fan of adding a virtual function for that one window but I can't really think of any better solution at the moment.